app/src/main/java/org/mifos/mobile/ui/fragments/SettingsFragment.kt

```diff
@@ -132,7 +134,23 @@ class SettingsFragment : PreferenceFragmentCompat(), OnSharedPreferenceChangeLis
     override fun onSharedPreferenceChanged(p0: SharedPreferences?, p1: String?) {
         val preference = findPreference(p1)
         if (preference is ListPreference) {
-            LanguageHelper.setLocale(context, preference.value)
+            val isSystemLanguage =
+                (preference.value == resources.getStringArray(R.array.languages_value)[0])
+            val preferences = PreferenceManager.getDefaultSharedPreferences(context)
+            preferences.edit()
+                .putBoolean(context?.getString(R.string.default_system_language), isSystemLanguage)
+                .apply()
```

Comment:
we already have a PreferenceHelper class in our project, i dont think we will need this

Reply:
Changed to PreferenceHelper.